go/libraries/doltcore/sqle/dolt_diff_summary_table_function.go

```diff
@@ -84,16 +86,29 @@ func (ds *DiffSummaryTableFunction) FunctionName() string {
 	return "dolt_diff_summary"
 }
 
+func (ds *DiffSummaryTableFunction) commitsResolved() bool {
+	if ds.dotCommitExpr != nil {
+		return ds.dotCommitExpr.Resolved()
+	}
+	return ds.fromCommitExpr.Resolved() && ds.toCommitExpr.Resolved()
+}
+
 // Resolved implements the sql.Resolvable interface
 func (ds *DiffSummaryTableFunction) Resolved() bool {
 	if ds.tableNameExpr != nil {
-		return ds.fromCommitExpr.Resolved() && ds.toCommitExpr.Resolved() && ds.tableNameExpr.Resolved()
+		return ds.commitsResolved() && ds.tableNameExpr.Resolved()
 	}
-	return ds.fromCommitExpr.Resolved() && ds.toCommitExpr.Resolved()
+	return ds.commitsResolved()
 }
 
 // String implements the Stringer interface
 func (ds *DiffSummaryTableFunction) String() string {
+	if ds.dotCommitExpr != nil {
+		if ds.tableNameExpr != nil {
+			return fmt.Sprintf("DOLT_DIFF_SUMMARY(%s, %s)", ds.dotCommitExpr.String(), ds.tableNameExpr.String())
+		}
+		return fmt.Sprintf("DOLT_DIFF_SUMMARY(%s)", ds.dotCommitExpr.String())
+	}
 	if ds.tableNameExpr != nil {
 		return fmt.Sprintf("DOLT_DIFF_SUMMARY(%s, %s, %s)", ds.fromCommitExpr.String(), ds.toCommitExpr.String(), ds.tableNameExpr.String())
```

Comment:
(minor) I like that you checked for `ds.tableNameExpr` above to avoid printing the tablename when it's not specified. Would be nice to do the same treatment here.

Reply:
oops! nevermind... the last line of this function was cut off and I didn't see where it handled the two arg form and thought it was here. my bad!